easy-ui-react/src/CodeBlock/CodeBlock.test.tsx

```diff
@@ -42,7 +42,11 @@ describe("<CodeBlock />", () => {
     });
 
     const copyBtn = screen.getByRole("button", { name: /copy code/i });
-    await userClick(user, copyBtn);
+    await act(async () => {
+      await waitFor(async () => {
+        await user.click(copyBtn);
+      });
+    });
 
```

Comment:
**question**: is this just a `vitest` thing that we may sometimes need a `watiFor` inside of  an `act`?

Reply:
it's a good question. i'm actually not entirely sure. i found the solve on a github issue somewhere. i'm guessing it has to do with the custom `Promise.resolve()` that i had to create for the DOM `clipboard` override and ensuring it's waited for with vitest